pkg/sdkserver/sdkserver.go

```diff
@@ -275,21 +273,41 @@ func (s *SDKServer) updateState() error {
 
 	s.gsUpdateMutex.RLock()
 	gs.Status.State = s.gsState
+
+	// if we are setting the Reserved status, check for the duration, and set that too
+	if gs.Status.State == stablev1alpha1.GameServerStateReserved && s.gsReserveDuration != nil {
+		n := metav1.NewTime(time.Now().Add(*s.gsReserveDuration))
+		gs.Status.ReservedUntil = &n
+	} else {
+		gs.Status.ReservedUntil = nil
+	}
 	s.gsUpdateMutex.RUnlock()
 
 	_, err = gameServers.Update(gs)
 	if err != nil {
 		return errors.Wrapf(err, "could not update GameServer %s/%s to state %s", s.namespace, s.gameServerName, gs.Status.State)
 	}
 
+	message := "SDK state change"
 	level := corev1.EventTypeNormal
 	// post state specific work here
 	switch gs.Status.State {
 	case stablev1alpha1.GameServerStateUnhealthy:
 		level = corev1.EventTypeWarning
+	case stablev1alpha1.GameServerStateReserved:
+		s.gsUpdateMutex.RLock()
+		hasDuration := s.gsReserveDuration != nil
+		if hasDuration {
+			message += fmt.Sprintf(", for %v", s.gsReserveDuration.String())
+		}
+		s.gsUpdateMutex.RUnlock()
+		// unfortunate complexity due to mutex locking, resetReserveAfter has it's own mutex
```

Comment:
same nit about comment sentences.

Reply:
Its not clear to me why you need to lock that mutex.

Reply:
I've cleaned this up, much simpler locking now. PTAL.